mobile/src/main/java/rs/readahead/washington/mobile/media/MediaFileHandler.java

```diff
@@ -199,16 +200,24 @@ public static void exportMediaFile(Context context, VaultFile vaultFile, @Nullab
 
     public static Single<VaultFile> importPhotoUri(Context context, Uri uri, @Nullable String parentId) throws Exception {
         // Vault replacement methods
-        InputStream v_input = context.getContentResolver().openInputStream(uri); // original photo
-        Bitmap v_bm = modifyOrientation(BitmapFactory.decodeStream(v_input), v_input); // bitmap of photo
-        Bitmap v_thumb = ThumbnailUtils.extractThumbnail(v_bm, v_bm.getWidth() / 10, v_bm.getHeight() / 10); // bitmap of thumb
-
+        Boolean keepExif = Preferences.isKeepExif();
+        ByteArrayOutputStream v_image_jpeg_stream = new ByteArrayOutputStream();
         ByteArrayOutputStream v_thumb_jpeg_stream = new ByteArrayOutputStream();
+
+        InputStream inputStream = context.getContentResolver().openInputStream(uri); // original photo
+        Bitmap v_bm = modifyOrientation(BitmapFactory.decodeStream(inputStream), inputStream); // bitmap of photo
+        Bitmap v_thumb = ThumbnailUtils.extractThumbnail(v_bm, v_bm.getWidth() / 10, v_bm.getHeight() / 10); // bitmap of thumb
         v_thumb.compress(Bitmap.CompressFormat.JPEG, 100, v_thumb_jpeg_stream);
```

Comment:
- Let's use the try-with-resources statement to automatically close the input streams (inputStream and input) after use, reducing the need for explicit close calls.
- Combine the declaration and assignment of the inputStream variable within the try block for better scoping and resource handling.
- Reordered the code to improve readability and logical flow.
- Utilized try-with-resources for the input stream within the keepExif branch to automatically close the stream after use.

 ```
    boolean keepExif = Preferences.isKeepExif();
    ByteArrayOutputStream imageJpegStream = new ByteArrayOutputStream();
    ByteArrayOutputStream thumbJpegStream = new ByteArrayOutputStream();
 try (InputStream inputStream = context.getContentResolver().openInputStream(uri)) {
        Bitmap originalBitmap = modifyOrientation(BitmapFactory.decodeStream(inputStream), inputStream); // bitmap of photo
        Bitmap thumbnail = ThumbnailUtils.extractThumbnail(originalBitmap, originalBitmap.getWidth() / 10, originalBitmap.getHeight() / 10); // bitmap of thumb
        thumbnail.compress(Bitmap.CompressFormat.JPEG, 100, thumbJpegStream);

        if (keepExif) {
            try (InputStream input = context.getContentResolver().openInputStream(uri)) {
                copyStream(input, imageJpegStream);
            }
        } else {
            if (!originalBitmap.compress(Bitmap.CompressFormat.JPEG, 100, imageJpegStream)) {
                throw new Exception("JPEG compression failed");
            }
        }
    }
```